benchmarks/experiment_runner.py

```diff
@@ -284,7 +284,7 @@ def run_once_and_gather_metrics(self, benchmark_experiment, benchmark_model,
 
     # Reset state and sync.
     reset_rng_state(benchmark_experiment)
-    self._mark_step(benchmark_experiment)
+    self._mark_step(benchmark_experiment, inputs_list[0])
```

Comment:
I'm not familiar with `torch_xla2`, but shouldn't we `mark_step` on all elements of `inputs_list`?

Reply:
Oh yes, I think you are right!

Reply:
Hmm, just double checked, the inputs_list contains the inputs for all iterations. inputs_list[0] should be the one that feed into the model in the first step. However, I am not sure if we need to `block_until_ready` for all inputs. But to be safe, I will just block all inputs. WDYT?

Reply:
I think that works.
But, if we didn't, wouldn't we be also possibly timing the operations outside the i-th step of the model?

Reply:
Let's call `jax.block_until_ready` on all inputs.

Reply:
Yes, `_mark_step` internally is calling `block_until_ready` if using torch_xla2.